account_analytic_required/models/account.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2011-2020 Akretion - Alexis de Lattre
+# Copyright 2016-2020 Camptocamp SA
+# Copyright 2020 Druidoo - Iván Todorovich
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, exceptions, fields, models
+
+
+class AccountAccountType(models.Model):
+    _inherit = "account.account.type"
+
+    property_analytic_policy = fields.Selection(
+        selection=[
+            ("optional", "Optional"),
+            ("always", "Always"),
+            ("posted", "Posted moves"),
+            ("never", "Never"),
+        ],
+        string="Policy for analytic account",
+        default="optional",
+        company_dependent=True,
+        help=(
+            "Sets the policy for analytic accounts.\n"
+            "If you select:\n"
+            "- Optional: The accountant is free to put an analytic account "
+            "on an account move line with this type of account.\n"
+            "- Always: The accountant will get an error message if "
+            "there is no analytic account.\n"
+            "- Posted moves: The accountant will get an error message if no "
+            "analytic account is defined when the move is posted.\n"
+            "- Never: The accountant will get an error message if an analytic "
+            "account is present.\n\n"
+            "This field is company dependent."
+        ),
+    )
+
+
+class AccountAccount(models.Model):
+    _inherit = "account.account"
+
+    def _get_analytic_policy(self):
+        """Extension point to obtain analytic policy for an account"""
+        self.ensure_one()
+        return self.user_type_id.with_company(
+            self.company_id.id
+        ).property_analytic_policy
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _post(self, soft=True):
+        res = super()._post(soft=soft)
+        self.mapped("line_ids")._check_analytic_required()
+        return res
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _check_analytic_required_msg(self):
```

Comment:
Can you review the v14
This will fail if there is any analytic distribution.. 
https://github.com/OCA/account-analytic/blob/14.0/account_analytic_required/models/account.py#L61

Reply:
done